src/fs/types.ts

```diff
@@ -21,32 +23,32 @@ export interface Properties {
 }
 
 export interface Posix {
-  exists(path: DistinctivePath): Promise<boolean>
-  get(path: DistinctivePath): Promise<PuttableUnixTree | File | null>
-  mv(from: DistinctivePath, to: DistinctivePath): Promise<this>
-  rm(path: DistinctivePath): Promise<this>
+  exists(path: DistinctivePath<PartitionedNonEmpty<Partition>>): Promise<boolean>
+  get(path: DistinctivePath<Partitioned<Partition>>): Promise<PuttableUnixTree | File | null>
+  mv(from: DistinctivePath<PartitionedNonEmpty<Partition>>, to: DistinctivePath<PartitionedNonEmpty<Partition>>): Promise<this>
+  rm(path: DistinctivePath<PartitionedNonEmpty<Partition>>): Promise<this>
 
   resolveSymlink(link: SoftLink): Promise<File | Tree | null>
-  symlink(args: { at: DirectoryPath; referringTo: DistinctivePath; name: string }): Promise<this>
+  symlink(args: { at: DirectoryPath<PartitionedNonEmpty<Partition>>; referringTo: DistinctivePath<PartitionedNonEmpty<Partition>>; name: string }): Promise<this>
 
   // Directories
-  ls(path: DirectoryPath): Promise<Links>
-  mkdir(path: DirectoryPath, options?: MutationOptions): Promise<this>
+  ls(path: DirectoryPath<Partitioned<Partition>>): Promise<Links>
+  mkdir(path: DirectoryPath<PartitionedNonEmpty<Partition>>, options?: MutationOptions): Promise<this>
 
   // Files
   write(
-    path: DistinctivePath,
+    path: DistinctivePath<PartitionedNonEmpty<Partition>>,
```

Comment:
Can this be restricted further to a `FilePath`? Are there cases where it is possible to write to a directory?